libs/wingsdk/src/core/tree.ts

```diff
@@ -68,6 +69,37 @@ export interface DisplayInfo {
    * @default - no source information
    */
   readonly sourceModule?: string;
+
+  /**
+   * UI components to display for this resource.
+   * @default - no UI components
+   */
+  readonly uiComponents?: any[]; // UIComponent
+}
+
+/** @internal */
+export type UIComponent = UIField | UISection | UIButton;
+
+/** @internal */
+export interface UIField {
+  readonly kind: "field";
+  readonly label: string;
+  readonly handlerPath: string;
+  readonly refreshRate: number | undefined;
+}
+
+/** @internal */
+export interface UIButton {
+  readonly kind: "button";
+  readonly label: string;
+  readonly handlerPath: string;
```

Comment:
"Path" is an ambiguous anyway so I would just call this `handler` and add some docstring that explains this is the node path of a cloud function handler.